internal/sysfs/osfile.go

```diff
@@ -98,9 +98,32 @@ func (f *osFile) reopen() (errno experimentalsys.Errno) {
 	// Clear any create flag, as we are re-opening, not re-creating.
 	f.flag &= ^experimentalsys.O_CREAT
 
+	var (
+		offset int64
+		err    error
+	)
+
+	if isDir, errno := f.IsDir(); errno == 0 && !isDir {
+		offset, err = f.file.Seek(0, 1)
```

Comment:
...declare error here...
```suggestion
		var err error
		offset, err = f.file.Seek(0, 1)
```